contracts/schemes/JoinAndQuit.sol

```diff
@@ -0,0 +1,244 @@
+        uint256 indexed _refund
+    );
+
+    event RedeemReputation(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _beneficiary,
+        uint256 _amount);
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _decision);
+
+    struct Proposal {
+        bool accepted;
+        address proposedMember;
+        address funder;
+        uint256 funding;
+    }
+
+    mapping(bytes32=>Proposal) public proposals;
+    mapping(address=>uint256) public fundings;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
+    Avatar public avatar;
+    IERC20 public fundingToken;
+    uint256 public minFeeToJoin;
+    uint256 public memberReputation;
+    uint256 public fundingGoal;
+    uint256 public fundingGoalDeadLine;
+    uint256 public totalDonation;
+    string public constant FUNDED_BEFORE_DEADLINE_KEY = "FUNDED_BEFORE_DEADLINE";
+    string public constant FUNDED_BEFORE_DEADLINE_VALUE = "TRUE";
+
+    /**
+     * @dev initialize
+     * @param _avatar the avatar this scheme referring to.
+     * @param _votingMachine the voting machines address to
+     * @param _voteParams voting machine parameters.
+     * @param _fundingToken the funding token
+     * @param _minFeeToJoin minimum fee required to join
+     * @param _memberReputation the repution which will be allocated for members
+     * @param _fundingGoal the funding goal
+     * @param _fundingGoalDeadLine the funding goal deadline
+     */
+    function initialize(
+        Avatar _avatar,
+        IntVoteInterface _votingMachine,
+        bytes32 _voteParams,
+        IERC20 _fundingToken,
+        uint256 _minFeeToJoin,
+        uint256 _memberReputation,
+        uint256 _fundingGoal,
+        uint256 _fundingGoalDeadLine
+    )
+    external
+    initializer
+    {
+        require(_avatar != Avatar(0), "avatar cannot be zero");
+        avatar = _avatar;
+        votingMachine = _votingMachine;
+        voteParams = _voteParams;
+        fundingToken = _fundingToken;
+        minFeeToJoin = _minFeeToJoin;
+        memberReputation = _memberReputation;
+        fundingGoal = _fundingGoal;
+        fundingGoalDeadLine = _fundingGoalDeadLine;
+    }
+
+    /**
+    * @dev execution of proposals, can only be called by the voting machine in which the vote is held.
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _decision a parameter of the voting result, 1 yes and 2 is no.
+    */
+    function executeProposal(bytes32 _proposalId, int256 _decision)
+    external
+    onlyVotingMachine(_proposalId)
+    returns(bool) {
+        require(proposals[_proposalId].accepted == false);
+        require(proposals[_proposalId].proposedMember != address(0));
+        Proposal memory proposal = proposals[_proposalId];
+        // Check if vote was successful:
+        if (_decision == 1) {
+            proposals[_proposalId].accepted = true;
```

Comment:
meaning `proposals[_proposalId].accepted` could have been false before, when the decision was already taking? Where do you see that? 

Maybe `accepted` here menas `executed`?

Reply:
yes , it is false before.
this is for the proposal decision. 
the proposal can be executed though not accepted by the dao .